x-pack/plugins/security_solution/server/lib/detection_engine/rules/read_rules.ts

```diff
@@ -24,12 +24,13 @@ export const readRules = async ({
   rulesClient,
   id,
   ruleId,
+  includeLegacyId,
```

Comment:
Can you please add a JSDoc entry for this new param -- will be helpful in a version or two when looking at this and thinking 'now which legacyId is this one?' 😅 

That said -- do you know how long we'll need to support this `legacyId`? So long as the user/rule could've been coming from `7.x` I imagine? This also gets interesting if you've got alerts from however long ago (1+ years) in cold storage you need to review... 


What about if you export via Security Solution in `7.x` and re-import in `8.x`, do they end up with a `legacyId` on import, or does it bypass the 'upgrade' when going through our import API?


This all has me wondering if it'd be best to just query against alerts using `ruleId` -- we'd end up showing all the alerts for every known instance of this Rule, but that's probably not a bad thing, and could actually play well into UX enhancements around selecting different revisions of a rule through time. I'll go ahead and move this conversation over to the issue here: https://github.com/elastic/kibana/issues/118935

Reply:
> What about if you export via Security Solution in 7.x and re-import in 8.x, do they end up with a legacyId on import, or does it bypass the 'upgrade' when going through our import API?

With the "override" selected on import the legacyId will remain, but right now the export does not include the legacyId so if a user exports rules in 7.16, deletes them, upgrades to 8.0 and imports those rules, the legacyId will be null.

Since the focus of this PR was to make sure the alerts in the rule detail page are present despite the `_id` change, I feel like the changes around the export and find should be made in a PR separate from this one. It will require another change to alerting's rules client "find" operation to include the legacy id for each rule and changes to the `_find` route and other places where we call the rules client `find`. In general I am curious what the outcome of just querying alerts using the `ruleId` would yield and what the drawbacks are..